packages/dataclass-editor/src/DataClassEditor.tsx

```diff
@@ -41,7 +44,24 @@ function DataClassEditor(props: EditorProps) {
     return <PanelMessage icon={IvyIcons.ErrorXMark} message={`An error has occurred: ${error.message}`} />;
   }
 
-  return <DataClassMaster dataClassFields={data.data.fields} />;
+  const dataClass = data.data;
+  const dataClassFields = dataClass.fields;
+
+  const title = isEntityClass(dataClass) ? 'Entity Class Editor' : 'Data Class Editor';
+  let detailTitle = title;
+  if (selectedDataClassFieldIndex !== undefined && selectedDataClassFieldIndex < dataClassFields.length) {
+    const selectedDataClassField = dataClassFields[selectedDataClassFieldIndex];
+    detailTitle = 'Attribute - ' + selectedDataClassField.name;
+  }
```

Comment:
why does this code live here and not in the editor? Because of this you need to add more properties to the Editor component. 
Other way to solve this is by simply do not this abstraction to a new Editor component.